commit 8004dd965b13b01a96def054d420f6df7ff22d53
Author: Yinghai Lu <yhlu.kernel@gmail.com>
Date:   Mon May 12 17:40:39 2008 -0700

    x86: amd opteron TOM2 mask val fix
    
    there is a typo in the mask value, need to remove that extra 0,
    to avoid 4bit clearing.
    
    Signed-off-by: Yinghal Lu <yhlu.kernel@gmail.com>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

Backported to Debian's 2.6.26 by dann frazier <dannf@hp.com>

diff -urpN linux-source-2.6.26.orig/arch/x86/kernel/cpu/mtrr/generic.c linux-source-2.6.26/arch/x86/kernel/cpu/mtrr/generic.c
--- linux-source-2.6.26.orig/arch/x86/kernel/cpu/mtrr/generic.c	2008-08-11 22:55:59.000000000 -0600
+++ linux-source-2.6.26/arch/x86/kernel/cpu/mtrr/generic.c	2008-08-11 22:57:13.000000000 -0600
@@ -219,7 +219,7 @@ void __init get_mtrr_state(void)
 		tom2 = hi;
 		tom2 <<= 32;
 		tom2 |= lo;
-		tom2 &= 0xffffff8000000ULL;
+		tom2 &= 0xffffff800000ULL;
 	}
 	if (mtrr_show) {
 		int high_width;
diff -urpN linux-source-2.6.26.orig/arch/x86/pci/k8-bus_64.c linux-source-2.6.26/arch/x86/pci/k8-bus_64.c
--- linux-source-2.6.26.orig/arch/x86/pci/k8-bus_64.c	2008-08-11 22:55:59.000000000 -0600
+++ linux-source-2.6.26/arch/x86/pci/k8-bus_64.c	2008-08-11 22:57:13.000000000 -0600
@@ -384,7 +384,7 @@ static int __init early_fill_mp_bus_info
 	/* need to take out [0, TOM) for RAM*/
 	address = MSR_K8_TOP_MEM1;
 	rdmsrl(address, val);
-	end = (val & 0xffffff8000000ULL);
+	end = (val & 0xffffff800000ULL);
 	printk(KERN_INFO "TOM: %016lx aka %ldM\n", end, end>>20);
 	if (end < (1ULL<<32))
 		update_range(range, 0, end - 1);
@@ -478,7 +478,7 @@ static int __init early_fill_mp_bus_info
 		/* TOP_MEM2 */
 		address = MSR_K8_TOP_MEM2;
 		rdmsrl(address, val);
-		end = (val & 0xffffff8000000ULL);
+		end = (val & 0xffffff800000ULL);
 		printk(KERN_INFO "TOM2: %016lx aka %ldM\n", end, end>>20);
 		update_range(range, 1ULL<<32, end - 1);
 	}
